src/components/Avatar/Avatar.Image.tsx

```diff
@@ -1,68 +1,263 @@
-
-  const imageStyle = {
-    transition: `opacity ${animationDuration}ms ${getEasingTiming(
-      animationEasing
-    )}`,
-  }
+import { ImageWrapperUI, ImageUI, TitleUI } from './styles/Avatar.css'
+import { getAnimationProps } from './Avatar.utils'
 
-  const contentMarkup = (
-    <ImageWrapperUI className={componentClassName} style={imageStyle}>
-      <ImageUI className="c-Avatar__image" style={{ backgroundImage }}>
-        <div className="c-Avatar__name">
-          <VisuallyHidden>{name}</VisuallyHidden>
-          <img
-            alt={name}
-            onError={onError}
-            onLoad={onLoad}
-            src={image}
-            style={{ display: 'none' }}
-          />
-        </div>
-      </ImageUI>
-    </ImageWrapperUI>
-  )
-
-  return hasImage ? contentMarkup : title
+let cache = {}
+export const clearCache = () => {
+  cache = {}
 }
+// ... 29,134,155,206
+export interface Props {
+  animation: boolean
+  animationDuration: number
+  animationEasing: string
+  className?: string
+  src?: string[] | string
+  light: boolean
+  name: string
+  onError?: () => void
+  onLoad?: () => void
+  title: string | number
+}
+
+export interface State {
+  currentIndex: number
+  isLoading: boolean
+  isLoaded: boolean
+}
+
+export class AvatarImage extends React.PureComponent<Props, State> {
+  static defaultProps = {
+    animation: true,
+    animationDuration: 160,
+    animationEasing: 'ease',
+    src: null,
+    onError: noop,
+    onLoad: noop,
+    name: null,
+    title: null,
+    light: false,
+  }
+
+  sourceList: string[] = []
+
+  image: any
+
+  state = {
+    currentIndex: 0,
+    isLoading: false,
+    isLoaded: false,
+  }
+
+  constructor(props) {
+    super(props)
+
+    this.sourceList = this.srcToArray(this.props.src)
+
+    // check cache to decide at which index to start
+    for (let i = 0; i < this.sourceList.length; i++) {
+      // if we've never seen this image before, the cache wont help.
+      // no need to look further, just start loading
+      /* istanbul ignore else */
+      if (!(this.sourceList[i] in cache)) break
+
+      // if we have loaded this image before, just load it again
+      /* istanbul ignore else */
+      if (cache[this.sourceList[i]] === true) {
+        this.state = { currentIndex: i, isLoading: false, isLoaded: true }
+        return
+      }
+    }
+
+    this.state = this.sourceList.length
+      ? // 'normal' opperation: start at 0 and try to load
+        { currentIndex: 0, isLoading: true, isLoaded: false }
+      : // if we dont have any sources, jump directly to unloaded
+        { currentIndex: 0, isLoading: false, isLoaded: false }
+  }
+
+  srcToArray = src => (Array.isArray(src) ? src : [src]).filter(x => x)
+
+  onLoad = () => {
+    cache[this.sourceList[this.state.currentIndex]] = true
+    /* istanbul ignore else */
+    if (this.image) {
+      this.setState({ isLoaded: true })
+      this.props.onLoad && this.props.onLoad()
```

Comment:
I think because you have a default prop of `noop` for `onLoad`, the check is not needed.

Reply:
fixed in a55572bd